http/src/main/java/io/micronaut/http/filter/BaseFilterProcessor.java

```diff
@@ -52,10 +67,50 @@
 public abstract class BaseFilterProcessor<A extends Annotation> implements ExecutableMethodProcessor<A> {
     private final BeanContext beanContext;
     private final Class<A> filterAnnotation;
+    private final RequestBinderRegistry argumentBinderRegistry;
+    private static final Set<String> PERMITTED_BINDING_ANNOTATIONS = Set.of(
+        Body.class.getName(),
+        Header.class.getName(),
+        QueryValue.class.getName(),
+        CookieValue.class.getName(),
+        PathVariable.class.getName()
+    );
 
     public BaseFilterProcessor(BeanContext beanContext, Class<A> filterAnnotation) {
         this.beanContext = beanContext;
         this.filterAnnotation = filterAnnotation;
+        Optional<RequestBinderRegistry> requestBinderRegistry = beanContext.findBean(RequestBinderRegistry.class);
+        this.argumentBinderRegistry = new RequestBinderRegistry() {
+            @Override
+            public <T> Optional<ArgumentBinder<T, HttpRequest<?>>> findArgumentBinder(Argument<T> argument) {
+                Class<? extends Annotation> annotation = argument.getAnnotationMetadata().getAnnotationTypeByStereotype(Bindable.class).orElse(null);
+                if (annotation != null && PERMITTED_BINDING_ANNOTATIONS.contains(annotation.getName())) {
+                    if (annotation == Body.class) {
+                        return Optional.of((context, source) -> {
+                            if (source instanceof FullHttpRequest<?> fullHttpRequest && fullHttpRequest.isFull()) {
+                                ByteBuffer<?> contents = fullHttpRequest.contents();
+                                if (contents != null) {
+                                    Argument<T> t = context.getArgument();
+                                    if (t.isAssignableFrom(ByteBuffer.class)) {
```

Comment:
could also go through RawBodyReader for this

Reply:
advantage would be ByteBuf param support, but maybe that's too difficult to be worth it